bulldozer/merge.go

```diff
@@ -252,24 +255,27 @@ func attemptMerge(ctx context.Context, pullCtx pull.Context, merger Merger, meth
 		gerr, ok := errors.Cause(err).(*github.ErrorResponse)
 		if !ok {
 			logger.Error().Err(err).Msg("Failed to merge pull request")
-			return false, true
+			return "", false, true
 		}
 
 		switch gerr.Response.StatusCode {
 		case http.StatusMethodNotAllowed:
-			logger.Info().Msgf("Merge rejected due to unsatisfied condition: %q", gerr.Message)
-			return false, false
+			reason = fmt.Sprintf("Merge rejected due to unsatisfied condition: %q", gerr.Message)
+			logger.Error().Msg(reason)
+			return reason, false, false
 		case http.StatusConflict:
-			logger.Info().Msgf("Merge rejected due to being invalid: %q", gerr.Message)
-			return false, false
+			reason = fmt.Sprintf("Merge rejected due to being invalid: %q", gerr.Message)
```

Comment:
We should check if this is the status reported when two Bulldozer instances race each other and the loser tries to merge an already-merged PR. I think it would just be confusing if Bulldozer posted a comment in that case.